### PR TITLE
minor: fix cargo clippy some warning

### PR DESCRIPTION
**Pull request by jackwener**
## Which issue does this PR close?

Closes #.

## Rationale for this change

fix cargo clippy some warning

### CODE REVIEW

**Comment by SteveLauC** [comment]
I am curious why I didn't get any clippy warnings locally, then I realized that these lints haven't been added to Rust 1.76 :D

**Comment by SteveLauC** [comment]
The CI error is not related to this PR, I have them in my PR too

File: datafusion/common/src/parsers.rs [hunk 1]
```diff
@@ -54,16 +55,17 @@ impl FromStr for CompressionTypeVariant {
...
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
             Self::GZIP => "GZIP",
             Self::BZIP2 => "BZIP2",
             Self::XZ => "XZ",
             Self::ZSTD => "ZSTD",
             Self::UNCOMPRESSED => "",
         }
-        .to_string()
+        .to_string();
```
Comment (Dandandan) [inline datafusion/common/src/parsers.rs:67]: `to_string` might be avoidable now?

File: datafusion/sqllogictest/src/engines/postgres_engine/types.rs [hunk 2]
```diff
@@ -37,8 +38,8 @@ impl<'a> FromSql<'a> for PgRegtype {
     }
 }
 
-impl ToString for PgRegtype {
-    fn to_string(&self) -> String {
-        self.value.clone()
+impl Display for PgRegtype {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.value.clone())
```
Comment (Dandandan) [inline datafusion/sqllogictest/src/engines/postgres_engine/types.rs:43]: No clone needed?

File: datafusion/common/src/parsers.rs [hunk 3]
```diff
@@ -54,16 +55,16 @@ impl FromStr for CompressionTypeVariant {
     }
 }
 
-impl ToString for CompressionTypeVariant {
-    fn to_string(&self) -> String {
-        match self {
+impl Display for CompressionTypeVariant {
```
Comment (alamb) [inline datafusion/common/src/parsers.rs:58]: 👍

**Review by alamb — APPROVED** [review]
Looks great to me -- thank you @jackwener

**Pull request merged**